Tests/PinningTests.swift

```diff
@@ -0,0 +1,252 @@
+        let config: TrustConfiguration = [domainConfig]
+
+        XCTAssertEqual(config.authenticationDispositionForFailedValidation(forHost: defaultHost), .cancelAuthenticationChallenge)
+        XCTAssertEqual(config.authenticationDispositionForFailedValidation(forHost: secondaryHost), .performDefaultHandling)
+
+        let domainConfig2 = TrustConfiguration.DomainConfiguration(domain: defaultHost, enforced: false, pinningHashes: [Data()])
+        let config2 = TrustConfiguration(domainConfigurations: [domainConfig2])
+
+        XCTAssertEqual(config2.authenticationDispositionForFailedValidation(forHost: defaultHost), .performDefaultHandling)
+    }
+
+    func test_TrustConfiguration_enforcesOnlyBeforeExpiration() {
+        let expiration = Date().addingTimeInterval(100)
+        let domainConfig = TrustConfiguration.DomainConfiguration(domain: defaultHost, enforced: true, pinningHashes: [Data()], expirationPolicy: .allow(after: expiration))
+        let config = TrustConfiguration(domainConfigurations: [domainConfig])
+
+        XCTAssertTrue(config.shouldValidateCertificate(forHost: defaultHost, at: Date()))
+        XCTAssertFalse(config.shouldValidateCertificate(forHost: defaultHost, at: expiration.addingTimeInterval(1)))
+        XCTAssertFalse(config.shouldValidateCertificate(forHost: secondaryHost, at: expiration.addingTimeInterval(1)))
+        XCTAssertFalse(config.shouldValidateCertificate(forHost: secondaryHost, at: Date()))
+
+        let domainConfig2 = TrustConfiguration.DomainConfiguration(domain: defaultHost, enforced: true, pinningHashes: [Data()])
+        let config2 = TrustConfiguration(domainConfigurations: [domainConfig2])
+
+        XCTAssertTrue(config2.shouldValidateCertificate(forHost: defaultHost, at: Date()))
+        XCTAssertTrue(config2.shouldValidateCertificate(forHost: defaultHost, at: Date().addingTimeInterval(100)))
+        XCTAssertFalse(config2.shouldValidateCertificate(forHost: secondaryHost, at: Date()))
+    }
+    
+    func test_TrustConfiguration_properlyValidatesCertificates() {
+        guard let domainConfig = try? TrustConfiguration.DomainConfiguration(domain: defaultHost, certificates: [TestCertificates.google]) else {
+            return XCTFail("Unable to load testing certificate")
+        }
+        
+        XCTAssertTrue(domainConfig.validate(against: TestCertificates.google))
+        XCTAssertFalse(domainConfig.validate(against: TestCertificates.apple))
+    }
+    
+    func test_TrustConfiguration_properlyValidatesCertificateHashes() {
+        let domainConfig = TrustConfiguration.DomainConfiguration(domain: defaultHost, encodedPinningHashes: ["ivJZzhltgbIeXZGekPcWiLySsZ846YXSsGgyL9bjqEY="])
+        XCTAssertTrue(domainConfig.validate(against: TestCertificates.google))
+        XCTAssertFalse(domainConfig.validate(against: TestCertificates.apple))
+    }
+    
```

Comment:
I see the above 2 tests test the `TrustConfiguration.DomainConfiguration` initializers. Do we also need one more test to test the remaining initializer that takes `pinningHashes`?
```swift
init(domain: String, enforced: Bool = true, includeSubdomains: Bool = true, pinningHashes: [Data], expirationPolicy: CertificateExpirationPolicy = .block)
```

Reply:
I don't believe so as the `encodedPinningHashes` initializer runs through the `pinningHashes` initializer, just with the added benefit of it being easier to create input.